From 2250da93bfb6d6466e62fc719eaaa9662db4ff42 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juho=20V=C3=A4h=C3=A4-Herttua?= <juhovh@iki.fi>
Date: Mon, 20 Oct 2025 01:09:54 +1100
Subject: [PATCH 3/3] wgengine/router/osrouter: Do not add tailscale rule

Tailscale overriding the default routes can cause trouble in case
there are other VPNs enabled in the system. Removing it gives more
control over the routing rules.
---
 wgengine/router/osrouter/router_linux.go | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/wgengine/router/osrouter/router_linux.go b/wgengine/router/osrouter/router_linux.go
index 58bd0513a..a1d9f450c 100644
--- a/wgengine/router/osrouter/router_linux.go
+++ b/wgengine/router/osrouter/router_linux.go
@@ -1269,10 +1269,13 @@ func mustRouteTable(num int) RouteTable {
 	// (ie. with no fwmark set), this is the first routing table, so
 	// it takes precedence over all the others, ie. VPN routes always
 	// beat non-VPN routes.
-	{
-		Priority: 70,
-		Table:    tailscaleRouteTable.Num,
-	},
+	//
+	// FIXME: Removed in order to get control over default routes, see
+	//        https://github.com/juhovh/tailguard/issues/6 for info.
+	//{
+	//	Priority: 70,
+	//	Table:    tailscaleRouteTable.Num,
+	//},
 	// If that didn't match, then non-fwmark packets fall through to the
 	// usual rules (pref 32766 and 32767, ie. main and default).
 }
-- 
2.51.1

